WordPress/Sniffs/PHP/StrictInArraySniff.php

```
@@ -34,98 +44,61 @@ class WordPress_Sniffs_PHP_StrictInArraySniff implements PHP_CodeSniffer_Sniff {
-
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
 		// Check if the strict check is actually needed.
-		if ( false === $this->array_functions[ $token ] ) {
-			$hasComma = $phpcsFile->findPrevious( T_COMMA, ( $closeParenthesis - 1 ), ( $openParenthesis + 1 ) );
-			if ( false === $hasComma || end( $tokens[ $hasComma ]['nested_parenthesis'] ) !== $closeParenthesis ) {
+		if ( false === $this->target_functions[ $matched_content ] ) {
+			if ( count( $parameters ) === 1 ) {
 				return;
 			}
 		}
 
-		$errorData = array( $token );
-
-		if ( false === $lastToken ) {
-			$phpcsFile->addError( 'Missing arguments to %s.', $openParenthesis, 'MissingArguments', $errorData );
+		// We're only interested in the third parameter.
+		if ( false === isset( $parameters[3] ) || 'true' !== $parameters[3]['raw'] ) {
+			$this->phpcsFile->addWarning(
+				'Not using strict comparison for %s; supply true for third argument.',
+				( isset( $parameters[3]['start'] ) ? $parameters[3]['start'] : $parameters[1]['start'] ),
+				'MissingTrueStrict',
+				array( $matched_content )
+			);
 			return;
 		}
+	}
 
-		if ( T_TRUE !== $tokens[ $lastToken ]['code'] ) {
-			$phpcsFile->addWarning( 'Not using strict comparison for %s; supply true for third argument.', $lastToken, 'MissingTrueStrict', $errorData );
-			return;
-		}
-	} // End process().
+	/**
+	 * Process the function if no parameters were found.
+	 *
+	 * @since 0.11.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 *
+	 * @return void
+	 */
+	public function process_no_parameters( $stackPtr, $group_name, $matched_content ) {
+		$this->phpcsFile->addError( 'Missing arguments to %s.', $stackPtr, 'MissingArguments', array( $matched_content ) );
```

We do need this.
By far not all function calls have required parameters, so a function call without parameters can be perfectly valid code, no syntax errors, but might still need to throw an error under certain rules.